packages/polymath-ui/src/components/EthNetworkWrapper/actions.js

```diff
@@ -17,85 +32,119 @@ export type NetworkParams = {|
 // this has to be here for now
 const HARDCODED_NETWORK_ID = 15;
 
-export const CONNECTED = 'polymath-auth/CONNECTED';
+export const requestAuthorization = () => async (dispatch: Function) => {
+  try {
+    dispatch(fail(ERROR_ACCESS_REQUESTED));
```

Comment:
This one @Grsmto Is this ok? It seems like the request authorization dispatches an error in any case, and then if `enable` doesnt work it dispatches *another* action with `ERROR_ACCESS_DENIED`

Reply:
I understand how this is confusing. This is kind of an heritage from the previous code. As we have the polling `setInterval` that reloads the page when Metamask has changed, we don't need to have a success action triggered here because page would be reloaded anyway.
That's why there is only the error case handled here.
I refactored this btw and I'll add a comment why what I just explained.

Reply:
Awesome, thanks!